tips/TIP-0020/tip-0020.md

```diff
@@ -0,0 +1,497 @@
+
+The following table describes the entirety of a _Transaction Payload_ in its serialized form following the notation from [draft TIP-21](https://github.com/iotaledger/tips/pull/41):
+
+<table>
+  <tr>
+    <th>Name</th>
+    <th>Type</th>
+    <th>Description</th>
+  </tr>
+  <tr>
+    <td>Payload Type</td>
+    <td>uint32</td>
+    <td>
+      Set to <strong>value 6</strong> to denote a <i>TIP-20 Transaction Payload</i>.
+    </td>
+  </tr>
+  <tr>
+    <td valign="top">Essence <code>oneOf</code></td>
+    <td colspan="2">
+      <details open="true">
+        <summary>Transaction Essence</summary>
+        <blockquote>
+          Describes the essence data making up a transaction by defining its inputs, outputs and an optional payload.
+        </blockquote>
+        <table>
+          <tr>
+            <td><b>Name</b></td>
+            <td><b>Type</b></td>
+            <td><b>Description</b></td>
+          </tr>
+          <tr>
+            <td>Transaction Type</td>
+            <td>uint8</td>
+            <td>
+              Set to <strong>value 1</strong> to denote a <i>TIP-20 Transaction Essence</i>.
+            </td>
+          </tr>
+          <tr>
+            <td>Network ID</td>
+            <td>uint64</td>
+            <td>
+              The unique value denoting whether the block was meant for mainnet, shimmer, testnet, or a private networks.
+            </td>
+          </tr>
+          <tr>
+            <td>Inputs Count</td>
+            <td>uint16</td>
+            <td>The number of input entries.</td>
+          </tr>
+          <tr>
+            <td valign="top">Inputs <code>anyOf</code></td>
+            <td colspan="2">
+              <details>
+                <summary>UTXO Input</summary>
+                <blockquote>
+                  Describes an input which references an unspent transaction output to consume.
+                </blockquote>
+                <table>
+                  <tr>
+                    <td><b>Name</b></td>
+                    <td><b>Type</b></td>
+                    <td><b>Description</b></td>
+                  </tr>
+                  <tr>
+                    <td>Input Type</td>
+                    <td>uint8</td>
+                    <td>
+                      Set to <strong>value 0</strong> to denote an <i>UTXO Input</i>.
+                    </td>
+                  </tr>
+                  <tr>
+                    <td>Transaction ID</td>
+                    <td>ByteArray[32]</td>
+                    <td>The BLAKE2b-256 hash of the transaction payload containing the referenced output.</td>
+                  </tr>
+                  <tr>
+                    <td>Transaction Output Index</td>
+                    <td>uint16</td>
+                    <td>The output index of the referenced output.</td>
+                  </tr>
+                </table>
+              </details>
+            </td>
+          </tr>
+          <tr>
+            <td>Inputs Commitment</td>
```

Comment:
The rationale behind inputs commitment is never really explained in this TIP, maybe a small paragraph about it in the next section would be nice?

Reply:
Added an explanation